File: files/2026-04-05/CSEC 05042026.R
library(tidyverse)
library(readxl)

path <- "2026-04-05/Challenge 110.xlsx"
input1 <- read_excel(path, range = "B2:B3") %>% pull()
input2 <- read_excel(path, range = "C2:C3") %>% pull()
input3 <- read_excel(path, range = "D2:D3") %>% pull()

test1 <- read_excel(path, range = "B5:B22", col_names = FALSE) %>% pull()
test2 <- read_excel(path, range = "C5:C22", col_names = FALSE) %>% pull()
test3 <- read_excel(path, range = "D5:D22", col_names = FALSE) %>% pull()

while (length(test3) > 0 && is.na(tail(test3, n = 1))) {
  test3 <- head(test3, -1)
}

create_sequence <- function(input) {
  input = str_split(input, ",") %>% unlist() %>% as.numeric()
  sequence <- c()
  for (value in input) {
    if (value > 0) {
      sequence <- c(sequence, rep(1, value))
      sequence <- c(sequence, NA)
    } else {
      sequence <- c(sequence, NA, NA)
    }
  }
  if (length(sequence) > 0 && is.na(tail(sequence, n = 1))) {
    sequence <- head(sequence, -1)
  }
  return(sequence)
}

s1 = create_sequence(input1)
s2 = create_sequence(input2)
s3 = create_sequence(input3)

identical(s1, test1) # TRUE
identical(s2, test2) # TRUE
identical(s3, test3) # TRUE

File: files/2026-04-05/CSEC 05042026.py
import pandas as pd
import numpy as np

path = "2026-04-05/Challenge 110.xlsx"
input1 = pd.read_excel(path, usecols="B", nrows = 1, skiprows = 1).values
input2 = pd.read_excel(path, usecols="C", nrows = 1, skiprows = 1).values
input3 = pd.read_excel(path, usecols="D", nrows = 1, skiprows = 1).values

test1 = pd.read_excel(path, usecols="B", nrows = 18, skiprows = 3).values.flatten()
test2 = pd.read_excel(path, usecols="C", nrows = 18, skiprows = 3).values.flatten()
test3 = pd.read_excel(path, usecols="D", nrows = 18, skiprows = 3).values.flatten()
while test3.size > 0 and np.isnan(test3[-1]):
    test3 = test3[:-1]

def seq(str):
    nums = [int(x) for x in str.split(",")]
    result = []
    for num in nums:
        if num == 0:
            result.append(np.nan)
            result.append(np.nan)
        else:
            result.extend([1] * num)
            result.append(np.nan)
    while result and np.isnan(result[-1]):
        result.pop()
    return result

s1  = seq(input1[0][0])
s2 = seq(input2[0][0])
s3 = seq(input3[0][0])

def equal_with_nan(a, b):
    a = np.asarray(a, dtype=float)
    b = np.asarray(b, dtype=float)
    return a.shape == b.shape and np.array_equal(a, b, equal_nan=True)

print(equal_with_nan(s1, test1)) # True
print(equal_with_nan(s2, test2)) # True
print(equal_with_nan(s3, test3)) # True